content/en/docs/tasks/administer-cluster/change-pv-access-mode-readwriteoncepod.md

```diff
@@ -0,0 +1,171 @@
+---
+title: Change the Access Mode of a PersistentVolume to ReadWriteOncePod
+content_type: task
+weight: 90
+min-kubernetes-server-version: 1.22
+---
+
+<!-- overview -->
+This page shows how to change the access mode on an existing PersistentVolume to
+use ReadWriteOncePod.
+
+## {{% heading "prerequisites" %}}
+
+{{< include "task-tutorial-prereqs.md" >}} {{< version-check >}}
+
+{{< note >}}
+The ReadWriteOncePod access mode is only supported for
+{{< glossary_tooltip text="CSI" term_id="csi" >}} volumes.
+To use this volume access mode you will need to update the following
+[CSI sidecars](https://kubernetes-csi.github.io/docs/sidecar-containers.html)
+to these versions or greater:
+
+* [csi-provisioner:v3.0.0+](https://github.com/kubernetes-csi/external-provisioner/releases/tag/v3.0.0)
+* [csi-attacher:v3.3.0+](https://github.com/kubernetes-csi/external-attacher/releases/tag/v3.3.0)
+* [csi-resizer:v1.3.0+](https://github.com/kubernetes-csi/external-resizer/releases/tag/v1.3.0)
+{{< /note >}}
+
+## Why should I use ReadWriteOncePod?
+
+Prior to Kubernetes 1.22, the ReadWriteOnce access mode was commonly used to
+restrict PersistentVolume access for workloads that required single-writer
+access to storage. However, this access mode had a limitation: it restricted
+volume access to a single *node*, allowing multiple pods on the same node to
+read from and write to the same volume simultaneously. This could pose a risk
+for applications that demand strict single-writer access for data safety.
+
+If ensuring single-writer access is critical for your workloads, consider
+migrating your volumes to ReadWriteOncePod.
+
+<!-- steps -->
+
+## Migrating existing PersistentVolumes
+
+If you have existing PersistentVolumes, they can be migrated to use
+ReadWriteOncePod. Migrations are supported from any access mode to
+ReadWriteOncePod.
```

Comment:
I think this depends on the CSI driver? Some CSI driver relies on the type of access mode to decide whether it is block or file volume. ReadWriteMany indicates it is file volume. ReadWriteOnce indicates it is block volume. If the access mode can be changed from ReadWriteMany to ReadWriteOncePod and also from ReadWriteOnce to ReadWriteOncePod, then the driver won't know what type of volume it is any more.

Reply:
Ok thank you for clarifying. We can just focus on the migration from ReadWriteOnce since that's the most important / common case.

Reply:
Sure